operator/src/main/java/org/keycloak/operator/v2alpha1/WatchedSecretsStore.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    private Secret fetchExistingStore() {
+        return client.secrets().inNamespace(getNamespace()).withName(getName()).get();
+    }
+
+    private Secret getNewStore() {
+        return new SecretBuilder()
+                .withNewMetadata()
+                    .withName(getName())
+                    .withNamespace(getNamespace())
+                .endMetadata()
+                .build();
+    }
+
+    private Map<String, String> getNewLastObservedVersions() {
+        if (existingStore != null && existingStore.getData() != null) {
+            return existingStore.getData().entrySet().stream()
+                    .collect(Collectors.toMap(
+                            Map.Entry::getKey,
+                            e -> new String(Base64.getDecoder().decode(e.getValue()))
+                    ));
+        }
+        else {
+            return Collections.emptyMap();
+        }
+    }
+
+    private Map<String, String> getNewCurrentVersions() {
+        return currentSecrets.stream()
+                .collect(Collectors.toMap(s -> s.getMetadata().getName(), this::getSecretVersion));
+    }
+
+    private String getSecretVersion(Secret secret) {
+        String serializedData = Serialization.asYaml(secret.getData());
+        try {
+            // using hashes as it's more robust than resource versions that can change e.g. just when adding a label
+            byte[] bytes = MessageDigest.getInstance("MD5").digest(serializedData.getBytes(StandardCharsets.UTF_8));
+            return new BigInteger(1, bytes).toString(16);
+        }
+        catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Set<Secret> fetchCurrentSecrets(Set<String> secretsNames) {
+        return secretsNames.stream()
+                .map(n -> {
+                    Secret secret = client.secrets().inNamespace(getNamespace()).withName(n).get();
+                    if (secret == null) {
+                        throw new IllegalStateException("Secret " + n + " not found");
+                    }
+                    return secret;
+                })
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public String getName() {
+        return cr.getMetadata().getName() + STORE_SUFFIX;
+    }
+
+    public static EventSource getStoreEventSource(KubernetesClient client, String namespace) {
+        SharedIndexInformer<Secret> informer =
+                client.secrets()
+                        .inNamespace(namespace)
+                        .withLabel(Constants.TYPE_LABEL, TYPE)
+                        .runnableInformer(0);
+
+        return new InformerEventSource<>(informer, Mappers.fromOwnerReference()) {
+            @Override
+            public String name() {
+                return "watchedResourcesStoreEventSource";
+            }
+        };
+    }
+
+    public static EventSource getWatchedSecretsEventSource(KubernetesClient client, String namespace) {
```

Comment:
Curiosity, have you checked that the `eventSource`s are re-evaluated every time the reconcile loop is triggered?

Reply:
What do you mean? We're not saving a state or modifying the event sours at runtime.

Reply:
👍